packages/components/dropdowns/dropdown-menu/src/dropdown-menu.spec.tsx

```diff
@@ -74,4 +74,39 @@ describe('DropdownMenu', () => {
     });
     expect(await screen.findByText('Content')).not.toBeVisible();
   });
+
+  it('should pass data attributes without menuType', async () => {
+    await render(
+      <DropdownMenu
+        data-testid="dropdown-container"
+        triggerElement={
+          <SecondaryButton label="Trigger" data-testid="trigger-element" />
+        }
+      >
+        <DropdownMenu.ListMenuItem>Option 1</DropdownMenu.ListMenuItem>
+      </DropdownMenu>
+    );
+
+    // Check that the data-testid attributes are passed down
+    expect(screen.getByTestId('trigger-element')).toBeInTheDocument();
+    expect(screen.getByTestId('dropdown-container')).toBeInTheDocument();
+  });
+
+  it('should pass data attributes with menuType="list"', async () => {
+    await render(
+      <DropdownMenu
+        menuType="list"
+        data-testid="dropdown-container"
+        triggerElement={
+          <SecondaryButton label="Trigger" data-testid="trigger-element" />
+        }
+      >
+        <DropdownMenu.ListMenuItem>Option 1</DropdownMenu.ListMenuItem>
```

Comment:
@ddouglasz Not knowing how the tracking-mechanism works exactly, but would it make sense to let `<DropdownMenu.ListMenuItem>` forward data-attributes as well? After all, those are the elements with the `onClick` handlers.

Reply:
Good point @misama-ct  but it seems like the marketing team is interested in tracking the popularity of components as it seems. I am not so sure if we need to go a degree deeper to track the handlers. 
reference: https://github.com/commercetools/ui-kit/issues/2849

Reply:
I have updated it in this commit: https://github.com/commercetools/ui-kit/pull/2855/commits/e377d08dfe6b93961fc057ad1145c2b7e3ae0a7d. @ddouglasz and @misama-ct Can you please review it again. As DropdownListMenuItem was using AccessibleButton, I had to spread data attributes in accessible button component as well.

@jocelyn-dunkley Can you please verify through the test cases if this is the required functionality?